File: 프로그래머스/unrated/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    answer = []
    for idx, f in enumerate(flag):
        if f:
            answer += [arr[idx]]*2*arr[idx]
        else:
            for i in range(arr[idx]):
                if answer:
                    answer.pop()
    
    return answer
        

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < flag.length; i++) {
            if (flag[i]) {
                for (int j = 0; j < arr[i] * 2; j++) {
                    stack.push(arr[i]);
                }
            } else {
                for (int j = 0; j < arr[i]; j++) {
                    stack.pop();
                }
            }
        }

        return stack.stream().mapToInt(i -> i).toArray();
    }
}